adapters/repos/db/lsmkv/store.go

```diff
@@ -30,18 +30,23 @@ import (
 // Store groups multiple buckets together, it "owns" one folder on the file
 // system
 type Store struct {
-	dir           string
-	rootDir       string
-	bucketsByName map[string]*Bucket
-	logger        logrus.FieldLogger
-	metrics       *Metrics
-
-	cycleCallbacks *storeCycleCallbacks
+	dir     string
+	rootDir string
 
 	// Prevent concurrent manipulations to the bucketsByNameMap, most notably
 	// when initializing buckets in parallel
 	bucketAccessLock sync.RWMutex
-	bucketByNameLock map[string]*sync.Mutex
+	bucketsByName    map[string]*Bucket
+
+	logger  logrus.FieldLogger
+	metrics *Metrics
+
+	cycleCallbacks *storeCycleCallbacks
+
+	// Prevent concurrent manipulations to the same Bucket, specially if there is
+	// action on the bucket in the meantime.
+	bucketLevelLock sync.RWMutex
```

Comment:
a sync.Mutex may be enough for bucketLevelLock

Reply:
yea, workin on it :). still doing testing